case-lib/lib.sh

```diff
@@ -140,10 +140,10 @@ func_lib_start_log_collect()
         return 3
     fi
 
-    local loggerCmd="$SOFLOGGER $logopt -l $ldcFile -o $logfile"
-    dlogi "Starting $loggerCmd"
+    local loggerCmd=("$SOFLOGGER" "$logopt" -l "$ldcFile" -o "$logfile")
```

Comment:
Quoting is good, but why we need () here?

Reply:
The parenthesis make it an array.  I'm just not sure what the difference between [*] and [@] is, as far as I can tell/remember they retrieve all elements, so not sure why we've got two versions. (The globber "*" might be clearer to most)

Reply:
Found the answer:
` If the word is double-quoted, ${name[*]} expands to a single word with the value of each array member separated by the first character of the IFS special variable, and ${name[@]} expands each element of name to a sep- arate word. When there are no array members, ${name[@]} expands to nothing. If the double-quoted expansion occurs within a word, the expansion of the first parameter is joined with the beginning part of the original word, and the expansion of the last parameter is joined with the last part of the original word.`

Reply:
> Quoting is good, but why we need () here?

From https://mywiki.wooledge.org/BashGuide/Arrays

> Strings are without a doubt the most used parameter type. But they are also the most misused parameter type. It is important to remember that a string holds just one element. [...] The only safe way to represent multiple string elements in Bash is through the use of arrays. 

Shellcheck is very good at catching this mistake.

Scrolling down to "Expand elements"

> The syntax "${myfiles[@]}" is extremely important. It works just like "$@" does for the positional parameters: it expands to a list of words, with each array element as one word, no matter what it contains. Even if there are spaces, tabs, newlines, quotation marks, or any other kind of characters in one of the array elements, it'll still be passed along as one word to whichever command we're running.

Also in https://www.gnu.org/software/bash/manual/html_node/Shell-Parameter-Expansion.html and https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#tag_18_05_02

Reply:
Thank you for good answers. I forgot about "()" and lost there.

Reply:
Same, at first I saw it and thought subshell.

Reply:
Guys, I'm sorry I can't fix 50 years old shell syntax and make it clearer :-) Arrays and subshells are fairly basic bash features, I don't think this requires a comment.